### PR TITLE
Change clone to assign non-enumerables properties

### PR DESCRIPTION
**Pull request by felipe-gustavo**
This will solve this issue #3070, I also add if the user added non-enumerable properties in array, [this line](https://github.com/ramda/ramda/commit/2f8e135902115f9110b12027c20360ce436b2d83#diff-3e22b8fb0b7b9c9e80982d8493e966efR35)

### CODE REVIEW

**Comment by Bradcomp** [comment]
Thanks for the PR @felipe-gustavo!

Can you add in a test for this?

**Comment by felipe-gustavo** [comment]
done, man!

File: source/internal/_clone.js [hunk 1]
```diff
@@ -24,15 +24,15 @@ export default function _clone(value, refFrom, refTo, deep) {
     }
     refFrom[idx] = value;
     refTo[idx] = copiedValue;
-    for (var key in value) {
+    for (var key of Object.getOwnPropertyNames(value)) {
```
Comment (CrossEye) [inline source/internal/_clone.js:27]: Most of Ramda will work -- believe it or not -- with ES3.  But I'm certain some ES5 has slipped in.

After 1.0, we will skip that and move to more modern JS.

While I like this change, It's not clear whether we should do it now or after 1.0.

File: test/clone.js [hunk 2]
```diff
@@ -212,3 +212,61 @@ describe('Let `R.clone` use an arbitrary user defined `clone` method', function(
   });
 
 });
+
+describe('Clone object with non-enumerable properties in Object and Array, preserve constructor', function() {
+
+  it('clone non-enumerable in object', function() {
+
+    function EfYP7(data) {
```
[the remark this answers is not in the document]
Comment (felipe-gustavo) [reply test/clone.js:220]: Doesn't invoke anything, I just work with random function name to tests, I'm sorry for paradigm break

File: source/internal/_clone.js (hunk 1 above)
Comment (felipe-gustavo) [reply source/internal/_clone.js:27]: Thanks for advice, I didn't read the entire documentation(in my language is the last note [pt-br](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyNames)), but just [`Object.create`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create) resolve this issue, I also noticed that this is already in production (#3067), so, can you close this PR and issue(#3070)?

Comment (felipe-gustavo) [reply source/internal/_clone.js:27]: Thanks for advice, I didn't read the entire documentation(in my language is the last note [pt-br](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyNames)), but just [`Object.create`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create) resolve this issue, I also noticed that this is already in production (#3067), so, I'm closing this PR

**Pull request closed without merging**